service/storage_service.cc

```diff
@@ -590,6 +590,7 @@ void storage_service::join_token_ring(int delay) {
     replicate_to_all_cores().get();
     // start participating in the ring.
     set_gossip_tokens(_bootstrap_tokens, _cdc_streams_ts);
```

Comment:
Note that this can potentially perform a write:
```
            _cdc_streams_ts = cdc::make_new_cdc_generation(db().local().get_config(),
```
(line 574), e.g. if we're doing a rolling upgrade. So the node might need to coordinate a write to other nodes. Can it do it if other nodes didn't see this node in gossip (even with status UNKNOWN)?

btw. this should only happen either during rolling upgrade (and perhaps replace if the replaced node didn't have a cdc generation timestamp, but I don't know if that ever actually happens), so the other nodes should know something about this node from an earlier time, even though now it is in shadow. I.e. they should have some status of this node; in case of rolling upgrade it will probably be DN.
So the nodes would receive a write from a DN node. I wonder if that's ok.

Could you perform the following test with this change:
1. bootstrap two node cluster with cdc experimental flag disabled
2. stop one node, enable cdc flag, restart
The restarted node should try to create a CDC generation during this long shadow round and send it to both nodes.

Reply:
Correction: it will only try to create a CDC generation if the following check passes:
```
bool should_propose_first_generation(const gms::inet_address& me, const gms::gossiper& g) {
    auto my_host_id = g.get_host_id(me);
    auto& eps = g.get_endpoint_states();
    return std::none_of(eps.begin(), eps.end(),
            [&] (const std::pair<gms::inet_address, gms::endpoint_state>& ep) {
        return my_host_id < g.get_host_id(ep.first);
    });
}
```

so we can be sure that one of the two nodes will attempt this.

So the test should be as follows:
1. bootstrap two node cluster with cdc experimental flag disabled
2. stop one node, enable cdc flag, restart
3. stop second node, enable cdc flag, restart

the write will happen in either step 2 or 3 depending on how the host ids were generated

Reply:
also my question is whether this check is now at all correct... does the node have its own endpoint state while in shadow? This comment:
> Initially, this gossiper will stay silent about what he knows, including himself

if I understand correctly, it means that it won't even know about itself?